### PR TITLE
JSRS Migration

### PR DESCRIPTION
**Pull request by bispodv**
Goals verified after changes:

- Make reservation
    - "Book a space resort on Pluto"
    - "Book a space resort"
    - "Book a space resort"-> The mercurial -> Book -> Select time period -> Honeymoon -> Yes
    - "Book a space resort with zero gravity" -> Robotel -> Book -> Select time period -> The Lab -> "Change that to next weekend" -> Yes
    - "make a reservation at a space resort with a spa for two astronauts" -> The Grand Titan -> Book -> Select time period -> The Lab -> "change my email to daniel.bispo@samsung.com" -> Yes
- SpaceResort
    - "Find space resorts"
    - "Find me a space resort close to Earth with a spa"
    - "Find space resorts" -> "on Jupiter"
    - "Find kid friendly resorts near Pluto" -> "with zero gravity"
    - "I need a space resort near Jupiter that has a refueling station"

All resulted in equal intents and equal IDE simulator results after migration comparing to before the migration

### CODE REVIEW

**Comment by bispodv** [comment]
@kangawoo44 @abapayeViv review, pls :)

File: capsule.bxb [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
+    }
+  }
+  store-sections {
+    section (TravelAndTransportation)
+  }
+  store-countries {
+    all
+  }
+  permissions {
+    library-permission (self:profile) // Note: While this permission is declared, it is not being used and the user will not be prompted, because we are fetching an 'imaginary' mocked profile from viv.self.  // This is done by passing an 'imaginary' flag to the self.GetProfile action, which we do on line 17 of models/actions/book/user-goals/CreateOrder.model.bxb: `value: self.Imaginary(true)` // Commenting out that line will cause viv.self to fetch the user's profile instead of a mocked profile, and the user will be prompted for this 'profile' permission when viv.self tries to fetch it.
```
Comment (kangawoo44) [inline capsule.bxb]: let's put the notes in a comment block so auto formatting doesn't do this.

File: code/find/lib/main.js [hunk 2]
```diff
@@ -1,77 +1,100 @@
-const data = require("./spaceResorts.js")
-const sorts = require("./sorts.js")
+import data from "./spaceResorts.js";
+import sorts from "./sorts.js";
 
-module.exports = {
-  "findSpaceResorts": findSpaceResorts,
-  "selectSpaceResorts": selectSpaceResorts
-}
+export default {
```
Comment (kangawoo44) [inline code/find/lib/main.js]: can we export functions individually?

**Comment by kangawoo44** [comment]
manual change: importing the polyfill dates library and using it in CreateItem.js and MakeReservation.js

Is there anything else you encountered?

**Comment by bispodv** [comment]
> manual change: importing the polyfill dates library and using it in CreateItem.js and MakeReservation.js
> 
> Is there anything else you encountered?

Those were the only uses for the old dates library in the JS files

File: code/find/lib/main.js (hunk 2 above)
Comment (bispodv) [reply code/find/lib/main.js]: @kangawoo44  Yes, I have just updated the PR with individual exports and block comment for capsule.bxb

**Comment by bispodv** [comment]
No warnings are shown in IDE

File: code/find/lib/main.js [hunk 3]
```diff
@@ -1,12 +1,6 @@
 import data from "./spaceResorts.js";
-import sorts from "./sorts.js";
 
-export default {
-  findSpaceResorts: findSpaceResorts,
-  selectSpaceResorts: selectSpaceResorts,
-};
-
-function findSpaceResorts(name, planet, searchCriteria, $vivContext) {
+export default function findResorts(name, planet, searchCriteria, $vivContext) {
```
Comment (kangawoo44) [inline code/find/lib/main.js]: why rename? can we keep the original name?

File: capsule.bxb [hunk 4]
```diff
@@ -33,6 +33,14 @@ capsule {
...
+    // Note: While this permission is declared, it is not being used
+    // and the user will not be prompted, because we are fetching an
+    // 'imaginary' mocked profile from viv.self.  // This is done by
+    // passing an 'imaginary' flag to the self.GetProfile action, which
+    // we do on line 17 of models/actions/book/user-goals/CreateOrder.model.bxb: `value: self.Imaginary(true)`
+    // Commenting out that line will cause viv.self to fetch the user's
+    // profile instead of a mocked profile, and the user will be prompted
+    // for this 'profile' permission when viv.self tries to fetch it.
+    library-permission (self:profile)
```
Comment (kangawoo44) [inline capsule.bxb:44]: oh I meant like this. If the current comments stay that way even with reformat function, it's fine.
/*
  comments
*/

Comment (bispodv) [reply capsule.bxb:44]: @kangawoo44 I tried that, but unfortunately those does not work in bxb files :( 

image

File: code/find/lib/main.js (hunk 3 above)
Comment (bispodv) [reply code/find/lib/main.js]: @kangawoo44 the problem was that the function was being used in another file, and it had the same name. I can rename the function that is calling "findSpaceResorts" instead, like in the image. This way there is no need to rename


image

File: capsule.bxb (hunk 4 above)
Comment (kangawoo44) [reply capsule.bxb:44]: oh I forgot these were bxb files 😂

File: code/find/lib/main.js (hunk 3 above)
Comment (kangawoo44) [reply code/find/lib/main.js]: Sure that seems fine. Another thing we could do is we don't declare a default export for main.js and import all as `main` and keep the same syntax as before

Comment (bispodv) [reply code/find/lib/main.js]: @kangawoo44 Seems like we are not allowed to leave the file without a default export:
image

Comment (kangawoo44) [reply code/find/lib/main.js]: oh I see. we can still import all as main.

Comment (bispodv) [reply code/find/lib/main.js]: Yes, importing everything as main now

Comment (kangawoo44) [reply code/find/lib/main.js]: I liked your idea too btw 😄

File: code/find/FindSpaceResorts.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import findResorts from "./lib/main.js";
+import main from "./lib/main.js";
```
Comment (kangawoo44) [inline code/find/FindSpaceResorts.js]: I think it should be
```
import * as main
```
no?

Comment (bispodv) [reply code/find/FindSpaceResorts.js]: Yes, you are correct. Works fine that way, thanks for commenting 
image

**Review by kangawoo44 — APPROVED** [review]
LGTM

**Pull request merged**